src/eBot/Application/Application.php

```diff
@@ -34,16 +35,17 @@ public function run() {
         Logger::log($this->getName());
 
         // Initializing database
-        $this->initDatabase();
+		$this->mysqli_link = $this->initDatabase();
```

Comment:
Insted of do that and passing it through all the methods, I would create a DAO object to make easier change the driver if needed.